File: 1201-1300/1299-Replace Elements with Greatest Element on Right Side/1299-Replace Elements with Greatest Element on Right Side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        vector<int> result(arr.size(), -1);
        int maxRight = arr.back();
        int n = arr.size();
        for (int i = n - 2; i >= 0; --i) {
            result[i] = maxRight;
            maxRight = max(maxRight, arr[i]);
        }
        return result;
    }
};

File: 1201-1300/1299-Replace Elements with Greatest Element on Right Side/1299-Replace Elements with Greatest Element on Right Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        result = [-1] * len(arr)
        maxRight = arr[-1]
        for i in range(len(arr) - 2, -1, -1):
            result[i] = maxRight
            maxRight = max(maxRight, arr[i])
        return result
